docs/source/command_line.rst

```diff
@@ -49,6 +49,13 @@ for full details, see :ref:`running-mypy`.
     Asks mypy to type check the provided string as a program.
 
 
+.. option:: --exclude
+
+    A regular expression that matches file names, directory names and paths
+    which mypy should ignore while recursively discovering files to check.
+    Use forward slashes on all platforms.
```

Comment:
Add one or two examples of typical use cases, such as ignoring a `build` or `venv` directory (venvs can contain `.py` files outside `site-packages` that users may want to exclude). Another example might be `/setup.py$` to avoid duplicate `setup.py` files.

Maybe also mention which names are always excluded?

Mention that giving a path (or `-p`/`-m`?) overrides exclusions, and give an example, such as `mypy site-packages/example/*.py`).